src/config.rs

```diff
@@ -82,6 +84,24 @@ impl Config for FsConfig {
         let shell_path = self.root_path.join("shells").join(name);
         shell_path.is_dir()
     }
+
+    fn get_shell_list(&self) -> io::Result<Vec<String>> {
+        let mut shell_names = Vec::new();
+        let root_path = self.shell_root_path().to_owned();
+        if try!(fs::metadata(root_path.to_owned())).is_dir() {
```

Comment:
we probably don't need either of these calls to `to_owned`.  Instead, pass the `root_path` as a borrow like so: `&root_path`